File: Codigos estudiantes por lenguaje/Java/Cristian Maldonado/Geometria/Line_Through_Two_Points.java
public class Line_Through_Two_Points {

    public static void main(String[] args) {
        Pair P = new Pair(3, 2);
        Pair Q = new Pair(2, 6);
        lineFromPoints(P, Q);
    }

    static void lineFromPoints(Pair P, Pair Q) {
        double a = Q.second - P.second;
        double b = P.first - Q.first;
        double c = a * (P.first) + b * (P.second);
        if (b < 0) {
            System.out.println("La linea que pasa a traves de "
                    + "los puntos P y Q es: "
                    + a + "x " + b + "y = " + c);
        } else {
            System.out.println("La linea que pasa a traves de l"
                    + "os puntos P y Q es: "
                    + a + "x + " + b + "y = " + c);
        }
    }

    static class Pair {

        int first;
        int second;

        public Pair(int first, int second) {
            this.first = first;
            this.second = second;
        }
    }
}

File: Codigos estudiantes por lenguaje/C++/Cristian Maldonado/Geometria/Line_Through_Two_Points.cpp
#include <iostream>
//-----------------//
#define x first
#define y second

using namespace std;

typedef pair<int,int> Pair;

void lineFromPoints(Pair p, Pair q){
    double a = q.y - p.y;
    double b = p.x - q.x;
    double c = a * (p.x) + b * p.y;
    if(b < 0){
        cout<<"La linea que pasa a traves de los puntos P y Q es:"<<endl;
        cout<<"("<<a<<"x) * ("<<b<<"y) = "<<c<<endl;
    }else{
        cout << "La linea que pasa a traves de los punto P y Q es:"<<endl;
        cout<<a<<"x + "<<b<< "y = "<<c<<endl;
    }
}

int main()
{
    ios_base::sync_with_stdio(false);cin.tie(NULL);cout.tie(NULL);
    int a, b; cin>>a>>b;
    int c, d; cin>>c>>d;
    Pair p = make_pair(a, b);
    Pair q = make_pair(c, d);
    lineFromPoints(p, q);
    return 0;
}
